src/Parsers/ParserDataType.cpp

```diff
@@ -78,14 +80,24 @@ bool ParserDataType::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     ++pos;
 
     /// Parse optional parameters
-    ParserList args_parser(std::make_unique<ParserExpression>(), std::make_unique<ParserToken>(TokenType::Comma));
     ASTPtr expr_list_args;
 
-    if (!args_parser.parse(pos, expr_list_args, expected))
-        return false;
-    if (pos->type != TokenType::ClosingRoundBracket)
-        return false;
-    ++pos;
+    ParserList args_parser_nested(std::make_unique<ParserNestedTable>(), std::make_unique<ParserToken>(TokenType::Comma), false);
+    if (args_parser_nested.parse(pos, expr_list_args, expected))
```

Comment:
Do we need to support mixed arguments list with `NestedTable` and `Expression`? I mean something like 
`Tuple(Tuple(key String, value String)), FixedString(16))`
where `Tuple(key String, value String)` cannot be parsed as `Expression` and `FixedString(16)` cannot be parsed as `NestedTable`.

Reply:
This is the case for CHYT, we have arbitrarily compound and mixed types.